test/common/http/conn_manager_impl_test_2.cc

```diff
@@ -263,6 +263,40 @@ TEST_F(HttpConnectionManagerImplTest, FrameFloodError) {
       StreamInfo::ResponseFlag::DownstreamProtocolError));
 }
 
+TEST_F(HttpConnectionManagerImplTest, EnvoyOverloadError) {
```

Comment:
Can we also have a test for what happens when the overload happens after the response has started?

Reply:
added tests for this.